src/PerfView/SymbolReaderHttpHandler.cs

```diff
@@ -0,0 +1,2010 @@
+        {
+            if (string.IsNullOrEmpty(output))
+            {
+                WriteStatusLog("The Git Credential Manager output was empty.");
+                return null;
+            }
+
+            // Example GCM output:
+            // protocol=https
+            // host=github.com
+            // path=
+            // username=<username>
+            // password=<password>
+            const string UserNameKeyName = "username=";
+            const string PasswordKeyName = "password=";
+            string username = null, password = null;
+            using (var reader = new StringReader(output))
+            {
+                string line;
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (string.IsNullOrWhiteSpace(line))
+                    {
+                        continue;
+                    }
+
+                    if (line.StartsWith(UserNameKeyName, StringComparison.Ordinal))
+                    {
+                        username = line.Substring(UserNameKeyName.Length);
+                    }
+                    else if (line.StartsWith(PasswordKeyName, StringComparison.Ordinal))
+                    {
+                        password = line.Substring(PasswordKeyName.Length);
+                    }
+                }
+            }
+
+            if (username != null && password != null)
+            {
+                // Try OAuth first and, if that fails, assume Basic auth.
+                return AuthToken.TryParseFromOAuthToken(username, password, out AuthToken token)
+                    ? token
+                    : AuthToken.CreateBasicFromUsernameAndPassword(username, password);
+            }
+
+            WriteStatusLog("Couldn't find a username and password in the output of Git Credential Manager.");
+            return null;
+        }
+
+        /// <summary>
+        /// Try to find the full path to the Git Credential Manager executable.
+        /// </summary>
+        /// <returns>The path, or null if one could not be found.</returns>
+        private static Task<string> LocateGcmExecutableAsync()
+        {
+            return Task.Run(() =>
+            {
+                foreach (string path in EnumeratePossibleGcmExecutableLocations())
+                {
+                    if (!string.IsNullOrEmpty(path) && File.Exists(path))
+                    {
+                        return path;
+                    }
+                }
+
+                return null;
+            });
+        }
+
+        /// <summary>
+        /// Try to locate the GCM executable in various places on disk.
+        /// </summary>
+        /// <returns>An enumeration of places to search. The values may not necessarily exist.</returns>
+        private static IEnumerable<string> EnumeratePossibleGcmExecutableLocations()
+        {
+            // The GCM_CORE_PATH environment variable.
+            yield return Environment.GetEnvironmentVariable("GCM_CORE_PATH");
```

Comment:
Can we wrap this in a call to App.IsElevated, or otherwise make this conditional, so that someone can't set this and then cause arbitrary code to run elevated?